subsys/mgmt/mcumgr/zephyr_grp/basic_mgmt.c

```diff
@@ -33,6 +36,89 @@ int storage_erase(void)
 	return rc;
 }
 
+
+#if defined(CONFIG_MCUMGR_GRP_BASIC_CMD_IMAGE_LIST)
+static int image_status(int id, char *buffer, ssize_t len)
+{
+	const struct flash_area *fa;
+	struct image_header hdr;
+	struct boot_swap_state bss;
+	int rc = flash_area_open(id, &fa);
+
+	if (rc < 0) {
+		LOG_ERR("Flash area %d open failed", id);
+		return -1;
+	}
+
+	rc = flash_area_read(fa, 0, &hdr, sizeof(hdr));
+	if (rc < 0) {
+		flash_area_close(fa);
+		LOG_ERR("Flash area %d read failed", id);
+		return -1;
+	}
+
+	rc = boot_read_swap_state(fa, &bss);
+	flash_area_close(fa);
+	if (rc < 0) {
+		LOG_ERR("Boot swap state %d read failed", id);
+		return -1;
+	}
+	if (hdr.ih_magic == IMAGE_MAGIC) {
+		snprintf(buffer, len, "ver=%d.%d.%d.%d%s", hdr.ih_ver.iv_major,
```

Comment:
I think for our need most suitable will be to introduce <b>the slot installation status field</b>. Perhaps of an integer type. Different statuses might be encoded as follow:
0 - no ongoing installation, also forced value for any slot which doesn't support installation. (Normally slots doesn't need to be installed). Also when read from application MCUmgr need to paste 0 always, as installation might be processed only while mcuboot is running.
1 - installation in progress, a positive range might be even used for reflect progress proportionally.
(-1) - installation error.

One installation is done value might return to 0.

The `copy_done` flag shouldn't be used to encode installation status as it has its own behavior defined in MCUboot, especially it is not used in overwrite-only mode at all.